### DISCUSSION TITLE
Allow specifying multiple config files

### DISCUSSION THREAD

**Discussion by chris-cmsoft** [Kong/kong#7294] [Ideas and feature requests]
### Summary

We are using Kong in Kubernetes as a Gateway rather than an ingress, as we need the route functionality as part of the gateway (Timeouts & rate limits are increases per route).

We have an automated process which registers routes from an OpenAPI spec, and the configuration for services in yaml can get fairly large. 

We are storing these yaml file as config maps, but Kubernetes has limits on how big the config maps can be. 

We thought we could possibly specify multiple yaml files for config, but not currently supported. 

Could be a great way for distributed teams to specify their own yaml files as well, without a big step in CI to generate a merged yaml file, and could be more declarative per team environment.

**Comment by p0pr0ck5** [comment]
+1, this would be very nice. Providing a config dir path allows for multiple teams to drop-in configs. Handling conflict might be a bit tricky, but there’s value there. It would also help in certain config management scenarios, eg chef/puppet/ansible drops in context-specific configs (eg a proxy.conf, admin.conf, db.conf, etc).

**Comment by chris-cmsoft** [comment]
Did some thinking around this, and providing a config dir path would be good, although will need to allow for recursive directories as well, as we will be specifying multiple configmaps for mounting inside the container, and these will probably be mounted on different sub paths.

```
# Example of mounting inside of kubernetes with configmaps
kong-configs/
├── service-x
│   └── kong.yml # Specified in configmap by service team
└── service-y
    └── kong.yml # Specified in configmap by service team
```

When the reloading or specifying configs to Kong, we would just pass the `kong-configs` dir as config path and it would need to find these files. 

But an initContainer on Kong would also serve the same purpose, which makes the possible solution simpler. 

Could be as simple as a cli command to merge configs and validate them and fail fast on conflict errors.

**Comment by mssanjay** [comment]
Does Kong support multiple config files or Directory now? This has become a stale topic and I am not sure if this option is currently available. 
I hate to use deck for manipulating config files. Why create additional complexity if kong can support multiple config files? We dont need to deck as additional step to do this merge. 
Appreciate your feedback. Thanks

**Comment by cello86** [comment]
We required our commercial person in Kong to have the possibility to use deck to create the declarative config file. This idea was proposed because deck provides the deduplication, defaults and tag management and it's can be useful in gitops environment to manage huge config situation

**Comment by pree04** [comment]
Hi, Is this feature available in kong ?. I have a usecase where I need to use multiple config maps in kong.
